test/conformance/context/urContextGetInfo.cpp

```diff
@@ -4,116 +4,147 @@
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urContextGetInfoTest);
 
-    std::unordered_map<ur_context_info_t, size_t> ctx_info_size_map;
-    std::unordered_map<ur_context_info_t, ur_memory_order_capability_flags_t>
-        ctx_info_mem_flags_map;
-};
-
-UUR_TEST_SUITE_P(urContextGetInfoTestWithInfoParam,
-                 ::testing::Values(
-
-                     UR_CONTEXT_INFO_NUM_DEVICES,                      //
-                     UR_CONTEXT_INFO_DEVICES,                          //
-                     UR_CONTEXT_INFO_USM_MEMCPY2D_SUPPORT,             //
-                     UR_CONTEXT_INFO_USM_FILL2D_SUPPORT,               //
-                     UR_CONTEXT_INFO_REFERENCE_COUNT,                  //
-                     UR_CONTEXT_INFO_ATOMIC_MEMORY_ORDER_CAPABILITIES, //
-                     UR_CONTEXT_INFO_ATOMIC_MEMORY_SCOPE_CAPABILITIES, //
-                     UR_CONTEXT_INFO_ATOMIC_FENCE_ORDER_CAPABILITIES,  //
-                     UR_CONTEXT_INFO_ATOMIC_FENCE_SCOPE_CAPABILITIES   //
-                     ),
-                 uur::deviceTestWithParamPrinter<ur_context_info_t>);
-
-TEST_P(urContextGetInfoTestWithInfoParam, Success) {
-    ur_context_info_t info = getParam();
-    size_t info_size = 0;
-    ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
-        urContextGetInfo(context, info, 0, nullptr, &info_size), info);
-    ASSERT_NE(info_size, 0);
+TEST_P(urContextGetInfoTest, SuccessNumDevices) {
+    ur_context_info_t info_type = UR_CONTEXT_INFO_NUM_DEVICES;
+    size_t size = 0;
 
-    if (const auto expected_size = ctx_info_size_map.find(info);
-        expected_size != ctx_info_size_map.end()) {
-        ASSERT_EQ(expected_size->second, info_size);
-    }
+    ASSERT_SUCCESS(urContextGetInfo(context, info_type, 0, nullptr, &size));
+    ASSERT_NE(size, 0);
+    ASSERT_EQ(size, sizeof(uint32_t));
 
-    std::vector<uint8_t> info_data(info_size);
+    uint32_t nDevices = 0;
     ASSERT_SUCCESS(
-        urContextGetInfo(context, info, info_size, info_data.data(), nullptr));
-
-    switch (info) {
-    case UR_CONTEXT_INFO_NUM_DEVICES: {
-        auto returned_num_of_devices =
-            reinterpret_cast<uint32_t *>(info_data.data());
-        ASSERT_GE(uur::DevicesEnvironment::instance->devices.size(),
-                  *returned_num_of_devices);
-        break;
-    }
-    case UR_CONTEXT_INFO_DEVICES: {
-        auto returned_devices =
-            reinterpret_cast<ur_device_handle_t *>(info_data.data());
-        size_t devices_count = info_size / sizeof(ur_device_handle_t);
+        urContextGetInfo(context, info_type, size, &nDevices, nullptr));
+
+    ASSERT_GE(uur::DevicesEnvironment::instance->devices.size(), nDevices);
+}
+
+TEST_P(urContextGetInfoTest, SuccessDevices) {
+    ur_context_info_t info_type = UR_CONTEXT_INFO_DEVICES;
+    size_t size = 0;
+
+    ASSERT_SUCCESS(urContextGetInfo(context, info_type, 0, nullptr, &size));
+    ASSERT_NE(size, 0);
+
+    if (size > 0) {
```

Comment:
I think we can just unconditionally do the part in here and get rid of the `else`, we've already checked that `size != 0` above